django_refugeerights/subscription/views.py

```diff
@@ -13,3 +18,26 @@ class SubscriptionViewSet(viewsets.ModelViewSet):
     queryset = Subscription.objects.all()
     serializer_class = SubscriptionSerializer
     filter_fields = ('contact_key', 'to_addr', 'active', )
+
+
+class SwitchSubscriptionView(APIView):
+    permission_classes = (permissions.IsAuthenticated,)
+
+    def post(self, request, format=None):
+        data = request.data
+        try:
+            contact_key = data['contact_key']
+        except KeyError:
+            raise ValidationError("A contact 'key' parameter is required.")
+        try:
+            messageset_id = data['messageset_id']
+        except KeyError:
+            raise ValidationError("A messageset 'id' parameter is required.")
+
+        messageset = MessageSet.objects.filter(pk=messageset_id).exists()
+        if messageset is not None:
```

Comment:
I think this should be `if not messageset:` (although perhaps `messageset` is a bad name for this variable because it's really `messageset_exists`).